sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -508,55 +277,89 @@ public String getLabel() {
     @JsonGetter("linkUrl")
     @JsonInclude(JsonInclude.Include.NON_NULL)
     @Nullable
-    public String getLinkUrl() {
-      return url;
+    public abstract String getLinkUrl();
+
+    private static <T> Item<T> create(String key, Type type, @Nullable T value) {
+      FormattedItemValue formatted = type.safeFormat(value);
+      return of(null, key, type, formatted.getLongValue(), formatted.getShortValue(), null, null);
     }
 
-    @Override
-    public String toString() {
-      return String.format("%s:%s=%s", ns, key, value);
+    /**
+     * Set the item {@link Item#getNamespace() namespace} from the given {@link Class}.
+     *
+     * <p>This method does not alter the current instance, but instead returns a new {@link Item}
+     * with the namespace set.
+     */
+    public Item<T> withNamespace(Class<?> namespace) {
```

Comment:
`ProxyInvocationHandler` uses `withNamespace` to register display data on specific `PipelineOptions` interfaces. I'd love to get rid of this if we can come up with a better design. `ProxyInvocationHandler` does crazy things to emulate Java classes.